subsys/net/ip/icmpv4.c

```diff
@@ -342,6 +340,19 @@ enum net_verdict net_icmpv4_input(struct net_pkt *pkt)
 		return NET_DROP;
 	}
 
+	if (net_is_ipv4_addr_bcast(net_pkt_iface(pkt),
+				   &NET_IPV4_HDR(pkt)->dst)) {
+		if (IS_ENABLED(CONFIG_NET_ICMPV4_ACCEPT_BROADCAST)) {
+			if (icmp_hdr.type != NET_ICMPV4_ECHO_REQUEST) {
+				NET_DBG("Dropping broadcast pkt");
```

Comment:
Why do we have 2 duplicate NET_DBG calls here? The only reason I can see is that you do don't want to break the condition in the `if` in 2 lines. And that's bad when codestyle issues direct us to write "bad" code (and we should increase line length to avoid premature wrapping (and decrease indent size)).

Reply:
There are two different if() cases here

1) if CONFIG_NET_ICMPV4_ACCEPT_BROADCAST is enabled AND if we do not have icmp echo-request message, then drop the msg
2) if CONFIG_NET_ICMPV4_ACCEPT_BROADCAST is not enabled, then drop the message right away

I do not see the major issue here.

Reply:
Sure, that can be written like that too. I will send a new version. BTW, next time you can suggest alternative way immediately, no need to ping-pong this around.

Reply:
Well, as I'm telling, this is a case of "2nd-order confusion". I know that you're experienced programmer, so start with the assumption that you didn't do that for a reason, which I don't understand.

Well, maybe it's not good approach every time, because it may lead to further levels of confusion.